server/odc-migrate/src/main/java/com/oceanbase/odc/migrate/jdbc/common/V42417HistoricalPartitionPlanMigrate.java

```diff
@@ -0,0 +1,310 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.oceanbase.odc.migrate.jdbc.common;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import javax.sql.DataSource;
+
+import org.apache.commons.collections4.CollectionUtils;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.transaction.support.TransactionTemplate;
+
+import com.oceanbase.odc.common.jpa.InsertSqlTemplateBuilder;
+import com.oceanbase.odc.common.json.JsonUtils;
+import com.oceanbase.odc.common.util.JdbcOperationsUtil;
+import com.oceanbase.odc.config.jpa.OdcJpaRepository.ValueGetterBuilder;
+import com.oceanbase.odc.core.migrate.JdbcMigratable;
+import com.oceanbase.odc.core.migrate.Migratable;
+import com.oceanbase.odc.core.shared.constant.TaskErrorStrategy;
+import com.oceanbase.odc.metadb.partitionplan.PartitionPlanEntity;
+import com.oceanbase.odc.metadb.partitionplan.PartitionPlanEntity_;
+import com.oceanbase.odc.metadb.partitionplan.PartitionPlanTableEntity;
+import com.oceanbase.odc.metadb.partitionplan.PartitionPlanTableEntity_;
+import com.oceanbase.odc.metadb.partitionplan.PartitionPlanTablePartitionKeyEntity;
+import com.oceanbase.odc.metadb.partitionplan.PartitionPlanTablePartitionKeyEntity_;
+import com.oceanbase.odc.plugin.task.api.partitionplan.datatype.TimeDataType;
+import com.oceanbase.odc.plugin.task.api.partitionplan.invoker.create.PartitionExprGenerator;
+import com.oceanbase.odc.plugin.task.api.partitionplan.invoker.partitionname.DateBasedPartitionNameGenerator;
+import com.oceanbase.odc.plugin.task.api.partitionplan.invoker.partitionname.PartitionNameGenerator;
+import com.oceanbase.odc.plugin.task.api.partitionplan.model.DateBasedPartitionNameGeneratorConfig;
+import com.oceanbase.odc.plugin.task.api.partitionplan.model.TimeIncreaseGeneratorConfig;
+import com.oceanbase.odc.service.partitionplan.model.PartitionPlanConfig;
+import com.oceanbase.odc.service.partitionplan.model.PartitionPlanStrategy;
+import com.oceanbase.odc.service.partitionplan.model.PartitionPlanTableConfig;
+
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * {@link V42417HistoricalPartitionPlanMigrate}
+ *
+ * @author yh263208
+ * @date 2024-03-11 17:07
+ * @since ODC_release_4.2.4
+ */
+@Slf4j
+@Migratable(version = "4.2.4.17", description = "migrate historical partition plan data")
+public class V42417HistoricalPartitionPlanMigrate implements JdbcMigratable {
+
+    @Override
+    public void migrate(DataSource dataSource) {
+        JdbcTemplate jdbcTemplate = new JdbcTemplate(dataSource);
+        List<PartitionPlanEntityWrapper> partitionPlanWrappers = generatePartitionPlanEntities(jdbcTemplate);
```

Comment:
may add some info level log output,
or we cannot trace from log if something unexpected happened

Reply:
okay, I have added several log printing